src/lib/src/common/workspace.ts

```diff
@@ -211,11 +211,11 @@ export class Workspace {
   }
 
   setTestStatus(path: string, status: ElementState) {
-    this.getWorkspaceElement(path).state = status;
+    this.setMarkerValue(path, 'testStatus', status);
   }
 
   getTestStatus(path: string): ElementState {
```

Comment:
True, there are still quite a few remnants remaining. The public interface of class `Workspace` definitely needs to be cleaned up. As a first step, I only changed the implementations behind the `Workspace` abstraction, so that things like `NavigationComponent`, which rely on it, would continue to work. Removing them will cause ripple effects of additional necessary changes that are better tackled in a separate step, imo.

That next step would be to cut out all the test-specific logic, e.g. the test status stuff in `NavigationComponent`, and put it into test-editor-web, instead. From there, the generic API of web-workspace-navigator can be used. For example, polling for test status currently remains implemented in `NavigationComponent`, but should in the future use the marker polling API that got introduced with https://github.com/test-editor/web-workspace-navigator/pull/54.